core/embed/rust/src/ui/model_mercury/component/footer.rs

```diff
@@ -2,27 +2,32 @@ use crate::{
     strutil::TString,
     ui::{
         component::{text::TextStyle, Component, Event, EventCtx, Never, SwipeDirection},
-        display::Color,
+        display::{Color, Font},
         event::SwipeEvent,
-        geometry::{Alignment, Offset, Rect},
+        geometry::{Alignment, Alignment2D, Offset, Point, Rect},
         lerp::Lerp,
         model_mercury::theme,
         shape,
         shape::{Renderer, Text},
     },
 };
 
-/// Component showing a task instruction (e.g. "Swipe up") and optionally task
-/// description (e.g. "Confirm transaction") to a user. A host of this component
-/// is responsible of providing the exact area considering also the spacing. The
-/// height must be 18px (only instruction) or 37px (both description and
-/// instruction). The content and style of both description and instruction is
-/// configurable separatedly.
+use heapless::String;
+
+/// Component showing a task instruction, e.g. "Swipe up", and optionally one of
+/// these:
+///     - a task description e.g. "Confirm transaction", or
+///     - a page counter e.g. "1 / 3", meaning the first screen of three total.
+/// A host of this component is responsible of providing the exact area
+/// considering also the spacing. The height must be 18px (only instruction) or
+/// 37px (instruction and description/position). The content and style of both
+/// is configurable separatedly.
 #[derive(Clone)]
 pub struct Footer<'a> {
     area: Rect,
     text_instruction: TString<'a>,
     text_description: Option<TString<'a>>,
+    counter: Option<PageCounter>,
```

Comment:
Nit: for the sake of structure size this should probably be an enum sharing space with `text_description`, `style_description` since both can't be used at once IIUC.

Reply:
reworked here https://github.com/trezor/trezor-firmware/pull/3954/commits/94305399464081fb22fe8efe170bf76e09a8149a 

I also removed `style_description` and `style_instruction` (and their setters) and made them `const` as they are consistent across all designs. The only reason why it was there was to support `PageCounter` color (green when max, e.g. `20 / 20`). But that's now handled internally by `PageCounter` component.